tests/rptest/tests/full_disk_test.py

```diff
@@ -130,6 +138,67 @@ def test_write_rejection(self):
                 expect_blocked=True)
             self.full_disk.clear_low_space()
 
+    @cluster(num_nodes=3, log_allow_list=FDT_LOG_ALLOW_LIST)
+    def test_reset_health_monitor(self):
+        """ Verify that health monitor frontend and backend have valid state after
+        they are reset by writing enough data that the disk health check triggers.
+        """
+        def check_health_monitor_frontend(disk_space_change: str):
+            # Looking for a log statement about a change in disk space.
+            # This is a check for the health monitor frontend because
+            # that structure logs disk space alerts.
+            pattern = f"Update disk health cache {disk_space_change}"
+            wait_until(
+                lambda: self.redpanda.search_log_any(pattern),
+                timeout_sec=5,
+                err_msg=f"Failed to find disk space change: {disk_space_change}"
+            )
+
+        def check_health_monitor_backend(fail_msg: str):
+            # This is an indirect check for the health monitor backend because
+            # RedpandaService.healthy() uses the prometheus metric, vectorized_cluster_partition_under_replicated_replicas,
+            # which is measured in the health monitor backend.
+            wait_until(lambda: self.redpanda.healthy(),
+                       timeout_sec=20,
+                       backoff_sec=1,
+                       err_msg=fail_msg)
+
+        self._setup()
+
+        self._send_all_topics("Q: Why should you be scared of computers?")
+        self.full_disk.trigger_low_space()
+        self._send_all_topics("A: They byte!", expect_blocked=True)
+
+        # Disk health should show degraded state
+        check_health_monitor_frontend(disk_space_change="ok -> degraded")
+
+        check_health_monitor_backend(
+            fail_msg="Cluster is not healthy before health monitor reset")
+
+        self.logger.debug("Check health monitor reset")
+        result_raw = self.admin.reset_health_info()
+        self.logger.debug(result_raw)
+        check_service_restart(self.redpanda,
+                              "Resetting Health Monitor Frontend")
+        check_service_restart(self.redpanda,
+                              "Resetting Health Monitor Backend")
+        assert result_raw.status_code == requests.codes.ok
+
+        check_health_monitor_backend(
+            fail_msg="Cluster is not healthy after health monitor reset")
+
+        # RP should still reject writes before clearing space
+        self._send_all_topics("Q: What animal does a computer like to watch?",
+                              expect_blocked=True)
+
+        self.full_disk.clear_low_space()
+
+        # RP should now accept writes after clearing space
+        self._send_all_topics("A: A RAM!")
+
+        # Expect the disk health check to tranistion from degraded to ok
+        check_health_monitor_frontend(disk_space_change="degraded -> ok")
```

Comment:
We might want to find out why the test passed before even though those caches weren't cleared

Reply:
Those are on the backend not frontend.
`_last_replies` I think can be ignored as it seems that structure is to keep track of broker health reports and it is not used to change data anywhere else. The code paths that I query in the DT tests do not touch this structure.

`_status` I think this can also be ignored as it gets cleared intermittently when users or other structures ask for a cluster health report. So I expect that this cache was cleared during test runtime, and thus did not affect test results.

`_reports` is the same. It gets cleared intermittently when users or other structures ask for a cluster health report. So I expect that this cache was cleared during test runtime, and thus did not affect test results.

Reply:
> Those are on the backend not frontend. `_last_replies` seems to be a structure the seed broker uses to log when the other brokers come up at startup. So failing to clear this structure did not produce a problem. The following log is from `` which is where `_last_replies` is used to get the health report for other brokers:
> 
> ```c++
>         vlog(
>           clusterlog.info,
>           "received node {} health report, marking node as up",
>           id);
> ```
> 
> And that long only printed during startup in my test.
> 
> `_status` this structure gets cleared intermittently when users or other structures ask for a cluster health report. So failing to clear it in the reset method is just another instance.
> 
> `_reports` this structure also gets cleared intermittently when users or other structures ask for a cluster health report. So failing to clear it in the reset method is just another instance.
> 
> I'll re-run the tests and confirm later tonight

A better reason is that before force-push `f06b0f3`, all the reset method did was re-register the leadership notification on the raft manager and abort in-flight refresh requests. So the reset method did not change the backend's state. Now we are actually changing the backend's state.